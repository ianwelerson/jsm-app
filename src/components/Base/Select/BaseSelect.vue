<script setup lang="ts">
export interface BaseSelectProps {
  name: string
  options: SelectOption[]
  selected?: string | null
}

export interface SelectOption {
  name: string
  value: string | number | null
  disabled?: boolean
}

const props = defineProps<BaseSelectProps>()
</script>

<template>
  <select :name="name" class="base-select">
    <option
      v-for="option in props.options"
      :key="option.value ?? ''"
      :value="option.value"
      :selected="option.value === props.selected"
      :disabled="option.disabled"
      class="base-select__option"
    >
      {{ option.name }}
    </option>
  </select>
</template>

<style lang="scss" scoped>
.base-select {
  appearance: none;
  -webkit-appearance: none;
  background: url("data:image/svg+xml,<svg width='12' height='8' xmlns='http://www.w3.org/2000/svg'><path d='M10.59.59 6 5.17 1.41.59 0 2l6 6 6-6L10.59.59Z'/></svg>")
    no-repeat right $transparent;
  border: none;
  color: $text-color;
  font-size: $text-sm;
  line-height: $leading-2;
  outline: none;
  padding-right: $spacing-2;
  position: relative;
  width: 100%;
}
</style>
