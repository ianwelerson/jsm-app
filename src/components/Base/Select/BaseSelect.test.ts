import { render } from '@testing-library/vue'
import BaseSelect from '@/components/Base/Select/BaseSelect.vue'

describe('BaseSelect', () => {
  it('should render the received options', async () => {
    const { getByText } = render(BaseSelect, {
      props: {
        name: 'base-select-test',
        options: [
          {
            name: 'Nome',
            value: 'name',
          },
          {
            name: 'Cidade',
            value: 'city',
          },
        ],
      },
    })

    getByText('Nome')
    getByText('Cidade')
  })

  it('should match the snapshot', async () => {
    const { html } = render(BaseSelect)

    expect(html()).toMatchSnapshot()
  })
})
