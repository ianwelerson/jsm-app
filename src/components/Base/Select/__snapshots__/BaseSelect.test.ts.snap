// Vitest Snapshot v1

exports[`BaseSelect > should match the snapshot 1`] = `"<select class=\\"base-select\\" data-v-bd273b9f=\\"\\"></select>"`;
